pybsn/__init__.py

```diff
@@ -15,6 +19,139 @@
 SCHEMA_PREFIX = "/api/v1/schema/"
 
 
+class _ClientTimeout:
+    pass
+
+
+"""Use the timeout that has been specified by the client."""
+CLIENT_TIMEOUT = _ClientTimeout()
+
+TimeoutOverride = Union[type(None), TimeoutSauce, type(CLIENT_TIMEOUT)]
+"""Type of argument to pass in requests, that indicates the time out behaviour
+   for waiting for a response.  None indicates to wait forever, TimeoutSauce
+   is the specified number of seconds, and CLIENT_TIMEOUT uses the default
+   value for the client.
+"""
+
+
+class TimeoutSession(requests.Session):
```

Comment:
It seems brittle to whitebox-extend a foreign class (`requests.Session`). This can break in arbitrary ways when `requests.Session` changes. In particular given that this is OSS repository, we should not be making assumptions of which version of requests is installed beyond the minimum requirements.

Much better to just handle the timeout ourselves in class `BigDBClient`. In particular, since all the requests are already shoehorned through a single method (`logged_request`) anyway, and you can handle the timeout there.